pkg/nmi/server/endpointpolicy.go

```diff
@@ -21,12 +21,12 @@ func ApplyEndpointRoutePolicy(podIP string, metadataIP string, metadataPort stri
 
 	endpoint, err := getEndpointByIP(podIP)
 	if err != nil {
-		return err
+		return fmt.Errorf("No endpoint found for Pod IP - %s. Error: %s", podIP, err.Error())
 	}
```

Comment:
newer version of Go support error wrapping where format specifier is %w and there is no need to use `err.Error()`.
Older version work very similar but without wrapping with %v specifier